src/components/alert/alert.tsx

```diff
@@ -318,22 +333,54 @@ export class Alert {
   };
 
   transitionEnd = (event: TransitionEvent): void => {
+    if (event.propertyName === this.activeTransitionProp) {
+      this.active ? this.openCloseEventEmitter("open") : this.openCloseEventEmitter("close");
+    }
+  };
+
+  transitionRun = (event: TransitionEvent): void => {
```

Comment:
In place of `this.transitionRunEvent()` you should be able to just decorate this function with `@Listen("transitionrun")` unless the `transitionrun` event doesn't bubble up to the Host element.  Does that work?

Reply:
Would be nice to rename that function to `onTransitionRun` to give it more semantic meaning too.

Reply:
When I use @Listen, it looks like the transitionRun bubbles but is not composed. According to MDN, all UA-dispatched UI events are composed (click/touch/mouseover/copy/paste, etc.). Most other types of events are not composed. This one looks like it bubbles, but is not composed.